crates/telio-traversal/src/session_keeper.rs

```diff
@@ -43,25 +50,41 @@ pub enum Error {
 
 type Result<T> = std::result::Result<T, Error>;
 
+#[derive(Debug, Eq, PartialEq, Clone)]
+pub enum AddNodeParameters {
+    NonBatching {
+        public_key: PublicKey,
+        target: dual_target::Target,
+        interval: Duration,
+    },
+    Batching {
+        public_key: PublicKey,
+        target: dual_target::Target,
+        interval: Duration,
+        threshold: Duration,
+    },
+}
```

Comment:
My way of choosing an explicit enum was fueled by the fact it should be very explicit what is being passed and where because of the friction batcher introduces - it should not harm anything, thus being very explicit about what's being passed makes sense to me.

Though I can understand that this duplication of code here might not be the nicest thing

Reply:
Threshold became an `Option<>`